### DISCUSSION TITLE
Push notifications

### DISCUSSION THREAD

**Discussion by logic-and-math** [supabase/supabase#1447] [Questions]
What would be the way to implement push notifications?
I guess I can write a server that listens to realtime service and then uses the firebase api to send the push notifications, but is there a built in way?
I guess also cloud functions will allow to send them.
Are there any plans to implement push notifications on triggers in a similar way it's done in firestore?

**Comment by henrykorir** [comment]
The realtime subscribe() does the job. You just have to build your notification UI.

**Comment by henrykorir** [comment]
@kiwicopple @awalias help on this one. How can push notifications be achieved with supabase?

**Reply by megacherry** [reply]
yeah, an answer or a roadmap would be awesome :)

@kiwicopple @awalias

**Reply by megacherry** [reply]
> 
> 
> @kiwicopple @awalias help on this one. How can push notifications be achieved with supabase?

i also found this issue: https://github.com/supabase/supabase/issues/1561

**Reply by w3b6x9** [reply]
We have discussed push notifications internally and considered building it as extensions to Realtime but no concrete plans yet.
